internal/aws/containerinsight/utils.go

```diff
@@ -177,6 +177,49 @@ func GetUnitForMetric(metric string) string {
 	return metricToUnitMap[metric]
 }
 
+type FieldsAndTagsPair struct {
+	Fields map[string]any
+	Tags   map[string]string
+}
+
+// ConvertToFieldsAndTags converts OTLP metric to a field containing metric values and a tag containing for decoration
+func ConvertToFieldsAndTags(m pmetric.Metric, logger *zap.Logger) []FieldsAndTagsPair {
+	var converted []FieldsAndTagsPair
+	if m.Name() == "" {
+		return converted
+	}
+
+	var dps pmetric.NumberDataPointSlice
+	switch m.Type() {
+	case pmetric.MetricTypeGauge:
+		dps = m.Gauge().DataPoints()
+	case pmetric.MetricTypeSum:
+		dps = m.Sum().DataPoints()
+	default:
+		logger.Warn("Unsupported metric type", zap.String("metric", m.Name()), zap.String("type", m.Type().String()))
+	}
+
+	if dps.Len() == 0 {
+		logger.Warn("Metric has no datapoint", zap.String("metric", m.Name()))
+	}
+
+	for i := 0; i < dps.Len(); i++ {
+		tags := make(map[string]string)
+		attrs := dps.At(i).Attributes()
+		attrs.Range(func(k string, v pcommon.Value) bool {
+			tags[k] = v.AsString()
+			return true
+		})
+		converted = append(converted, FieldsAndTagsPair{
+			Fields: map[string]any{
+				m.Name(): nil, // metric value not needed for attribute decoration
```

Comment:
If we're going to have this limitation (plus only supporting gauge and sum), then I think just move this helper method to the gpu decorator (it's only used from there right now anyway).  Having it here in this common place implies that it should work for any OTLP metric.